contracts/debtAllocators/GenericDebtAllocator.sol

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.18;
+
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+
+import {Governance} from "@periphery/utils/Governance.sol";
+import {IVault} from "../interfaces/IVault.sol";
+
+contract GenericDebtAllocator is Governance {
+    event SetTargetDebtRatio(
+        address indexed strategy,
+        uint256 targetRatio,
+        uint256 totalDebtRatio
+    );
+
+    event SetMinimumChange(address indexed strategy, uint256 minimumChange);
+
+    event SetMaxAcceptableBaseFee(uint256 maxAcceptableBaseFee);
+
+    // Struct for each strategies info.
+    struct Config {
+        uint256 targetRatio;
+        uint256 minimumChange;
+    }
+
+    uint256 internal constant MAX_BPS = 10_000;
+
+    // Mapping of strategy => its config.
+    mapping(address => Config) public configs;
+
+    // Address of the vault this serves as allocator for.
+    address public vault;
+
+    // Total debt ratio currently allocated in basis points.
+    // Can't be more than 10_000.
+    uint256 public debtRatio;
+
+    // Max the chains base fee can be during debt update.
+    // Will default to max uint256 and need to be set to be used.
+    uint256 public maxAcceptableBaseFee;
+
+    constructor(address _vault, address _governance) Governance(_governance) {
+        inizialize(_vault, _governance);
+    }
+
+    /**
+     * @notice Initializes the debt allocator.
+     * @dev Should be called atomiclly after cloning.
+     * @param _vault Address of the vault this allocates debt for.
+     * @param _governance Address to govern this contracts.
+     */
+    function inizialize(address _vault, address _governance) public {
+        require(address(vault) == address(0), "!initialized");
+        vault = _vault;
+        governance = _governance;
+        // Default max base fee to uint256 max
+        maxAcceptableBaseFee = type(uint256).max;
+    }
+
+    /**
+     * @notice Check if a strategies debt should be updated.
+     * @dev This should be called by a keeper to decide if a strategies
+     * debt should be updated and if so by how much.
+     *
+     * This cannot be used to withdraw down to 0 debt.
+     *
+     * @param _strategy Address of the strategy to check.
+     * @return . bool repersenting if the debt should be updated.
+     * @return . calldata that includes what the debt should be updated to.
```

Comment:
...and also returns a bytes encoded message in the `false` case

Reply:
https://github.com/yearn/vault-periphery/pull/15/commits/21b8ff41c6b5848a617675584c28b81077eb389e